### PR TITLE
Rename job to jobId in DeploymentGroup

### PR DESCRIPTION
**Pull request by davidxia**
To be consistent with other classes like Deployment

### CODE REVIEW

**Comment by davidxia** [comment]
@rohansingh @gimaker

**Comment by rohansingh** [comment]
:+1:

File: helios-client/src/main/java/com/spotify/helios/common/descriptors/DeploymentGroup.java [hunk 1]
```diff
@@ -55,30 +53,29 @@
...
-   * @param job The job for the deployment group.
+   * @param jobId The job ID for the deployment group.
    * @param hostSelectors The selectors that determine which agents are part of the deployment
    *                       group.
    */
   public DeploymentGroup(
       @JsonProperty("name") final String name,
       @JsonProperty("hostSelectors") final List<HostSelector> hostSelectors,
-      @JsonProperty("job") @Nullable final JobId job,
+      @JsonProperty("job") @Nullable final JobId jobId,
```
Comment (gimaker) [inline helios-client/src/main/java/com/spotify/helios/common/descriptors/DeploymentGroup.java]: should the json name also be jobId, then?

Comment (rohansingh) [inline helios-client/src/main/java/com/spotify/helios/common/descriptors/DeploymentGroup.java]: If we change it what will happen to existing stuff? We just clear out all the deployment groups we've created in testing?

Comment (gimaker) [inline helios-client/src/main/java/com/spotify/helios/common/descriptors/DeploymentGroup.java]: yes. if we don't, them those deployment groups will be defunct -- we might not even be able to remove them at that point.

Comment (davidxia) [inline helios-client/src/main/java/com/spotify/helios/common/descriptors/DeploymentGroup.java]: This is just internal variable naming. The json key will still be "job" like in Deployment.java

**Pull request merged**